### PR TITLE
GH-35537: [MATLAB] Create shared test class utility for numeric arrays

### LINKED ISSUE #35537: [MATLAB] Create shared test class utility for numeric arrays.
### Describe the enhancement requested

Many of the unit tests we'll write for numeric arrays will be identical. We should write a shared test class utility called `hNumericArray.m` that our test classes for numeric arrays - e.g. `tFloat64Arrray.m` - can inherit from. 

### Component(s)

MATLAB

### PR DESCRIPTION
**Pull request by sgilmore10**
### Rationale for this change
To avoid writing duplicate tests, we created an an Abstract Test Class called `hNumericArray` which defines tests that can be shared by all numeric array types. Concrete test classes can inherit from this class to automatically get the shared test points. 

### What changes are included in this PR?

1. Defined new abstract test class `hNumericArray`
2. Modified `tFloat32Array` to inherit from `hNumericArray`
3. Modified `tFloat64Array` to inherit from `hNumericArray`

### Are these changes tested?

1. Qualified the tests still pass on macOS.

### Are there any user-facing changes?
There are no user-facing changes in this PR. This PR only has test-related changes.

* Closes: #35537

### CODE REVIEW

File: matlab/test/arrow/array/hNumericArray.m [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+
+            % Create a Float64Array from an empty double vector
+            A3 = tc.ArrowArrayConstructor(tc.MatlabArrayFcn([]), DeepCopy=MakeDeepCopy);
+            data = toMATLAB(A3);
+            tc.verifyEqual(data, tc.MatlabArrayFcn(reshape([], 0, 1)));
+        end
+
+        function MatlabConversion(tc, MakeDeepCopy)
+        % Tests the type-specific conversion methods, e.g. single for
+        % arrow.array.Float32Array, double for array.array.Float64Array
```
Comment (kou) [inline matlab/test/arrow/array/hNumericArray.m:87]:
```suggestion
            % Tests the type-specific conversion methods, e.g. single for
            % arrow.array.Float32Array, double for array.array.Float64Array
```

**Review by kou — APPROVED** [review]
+1

**Comment by assignUser** [comment]
@kevingurney would you mind reviewing this?

**Comment by kou** [comment]
(I think that @sgilmore10 and @kevingurney did pair programming.)

**Comment by assignUser** [comment]
Ah, +1 then!

**Comment by kevingurney** [comment]
@assignUser and @kou - that's correct! @sgilmore10 and I have been pair programming on these MATLAB interface pull requests.

**Pull request merged**